command/pipelines/pipeline_execute.go

```diff
@@ -0,0 +1,99 @@
+package pipelines
+
+import (
+	"flag"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/spinnaker/spin/command"
+	gate "github.com/spinnaker/spin/gateapi"
+)
+
+type PipelineExecuteCommand struct {
+	ApiMeta command.ApiMeta
+
+	application string
+	name        string
+}
+
+// flagSet adds all options for this command to the flagset and returns the
+// flagset object for further modification by subcommands.
+func (c *PipelineExecuteCommand) flagSet() *flag.FlagSet {
+	cmd := "pipeline execute"
+
+	f := c.ApiMeta.GlobalFlagSet(cmd)
+	f.StringVar(&c.application, "application", "", "Spinnaker application the pipeline lives in")
+	f.StringVar(&c.name, "name", "", "Name of the pipeline to execute")
+
+	// TODO auto-generate flag help rather than putting it in "Help"
+	f.Usage = func() {
+		c.ApiMeta.Ui.Error(c.Help())
+	}
+
+	return f
+}
+
+// executePipeline calls the Gate endpoint to execute the pipeline.
+func (c *PipelineExecuteCommand) executePipeline() (gate.HttpEntity, *http.Response, error) {
+	entity, resp, err := c.ApiMeta.GateClient.PipelineControllerApi.InvokePipelineConfigUsingPOST1(nil,
```

Comment:
How does this work? I thought we still needed to send a pipeline config

Reply:
`Invoke` searches for a config based on app/name (like `get` does) and starts that config. We _can_ send an explicit config to run but that method may cause the pipeline execution not to be tracked in the UI, which doesn't seem like what we want.